popup/popup.js

```diff
@@ -1,5 +1,115 @@
-document.addEventListener('DOMContentLoaded', function() {
-    chrome.storage.local.get(['productData'], function(result) {
-        document.getElementById('data').innerText = JSON.stringify
+const proxyUrl = 'https://cors-anywhere.herokuapp.com/';
+const targetUrl = 'https://www.bing.com/api/shopping/v1/item/search?appid=67220BD2169C2EA709984467C21494086DF8CA85&features=persnlcashback&sf=cashback1';
+const url = proxyUrl + targetUrl;
+
+chrome.tabs.query({ active: true, currentWindow: true }, (tabs) => {
+    let tabUrl = new URL(tabs[0].url);
+    let domain = tabUrl.hostname;
+    let path = tabUrl.pathname;
+
+    if (domain.startsWith('www.')) {
+        domain = domain.substring(4);
+    }
+    
+    console.log(domain);
+    console.log(path);
+    const body = {
+        "AgeGroup": 0,  
+        "IsAADSignedIn": true,
+        "IsBingSignedInUser": false,
+        "IsMSASignedIn": false,
+        "IsOffTheRecord": false,
+        "IsPersonalizationDataConsentChanged": false,
+        "IsPersonalizationDataConsentEnabled": false,
+        "IsSSOEnabled": false,
+        "LdClickData": "",
+        "MsClickId": "",
+        "ReturnedToCashbackActivatedDomain": false,
+        "UClickData": "",
+        "item": {
+            "seller": {
+                "IsEdgeRebatesFlow": false,
+                "domain": domain,
+                "path": path
+            }
+        },
+        "sourceTypes": [
+            "deals"
+        ]
+    };
+    
+    async function postData(url = '', data = {}) {
+        const response = await fetch(url, {
+            method: 'POST',
+            headers: {
+                'Content-Type': 'application/json', 
+                'X-Requested-With': 'XMLHttpRequest'
+            },
+            body: JSON.stringify(data)
+        });
+        return response.json(); // parses JSON response into native JavaScript objects
+    }
+    
+    postData(url, body)
+        .then(data => {
+            console.log(data); // JSON data parsed by `response.json()` call
+            const allCheckoutPages = data.deals.retailerData.allCheckoutPages;
+            let matchingPage = allCheckoutPages.find(page => page.checkoutPageUrl === path);
```

Comment:
This can be more robust.
` let matchingPage = allCheckoutPages.find(page => page.checkoutPageUrl === path);`

1. Are we comparing the whole /checkout === https://github.com/Checkout? 
   If yes, then first ask this comparison will not work right?
2. What if the Url in retailersData is like:
    /checkout/buy
    and the Live site Url is /checkout/12345/buy
    Will you logic hold true here?